src/apiflask/app.py

```diff
@@ -1253,12 +1253,14 @@ def _add_response(
         if links is not None:
             operation['responses'][status_code]['links'] = links
         if headers_schema is not None:
-            headers = self._ma_plugin.converter.schema2parameters(  # type: ignore
+            header_params = self._ma_plugin.converter.schema2parameters(  # type: ignore
                 headers_schema, location='headers'
             )
-            operation['responses'][status_code]['headers'] = {
-                header['name']: header for header in headers
-            }
+            headers = {header['name']: header for header in header_params}
+            for header in headers.values():
+                header.pop('in', None)
+                header.pop('name', None)
```

Comment:
I don't think this should be happening for OAS 2.X, so I originally only performed this removal when `app.config['OPENAPI_VERSION']` was less than 3.0.0, but I discovered in testing that the generated spec for 2.0 failed the OAS validator for other reasons. I'm not sure what level of support this project is intending to provide for OAS < 3.0.0, so it seemed simplest to just make sure that this is working for 3.0.0+, but I'd be open to improving this to handle 2.X better.

Reply:
This project is intended to support only OAS >= 3.